.github/workflows/beam_PostCommit_TransformService_Direct.yml

```diff
@@ -0,0 +1,95 @@
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+name: PostCommit TransformService Direct
+
+on:
+  issue_comment:
+    types: [created]
+  schedule:
+    - cron: '0 */6 * * *'
+  workflow_dispatch:
+
+#Setting explicit permissions for the action to avoid the default permissions which are `write-all` in case of pull_request_target event
+permissions:
+  actions: write
+  pull-requests: read
+  checks: read
+  contents: read
+  deployments: read
+  id-token: none
+  issues: read
+  discussions: read
+  packages: read
+  pages: read
+  repository-projects: read
+  security-events: read
+  statuses: read
+
+# This allows a subsequently queued workflow run to interrupt previous runs
+concurrency:
+  group: '${{ github.workflow }} @ ${{ github.sha || github.head_ref || github.ref }}-${{ github.event.sender.login }}-${{ github.event.schedule }}'
+  cancel-in-progress: true
+
+env:
+  GRADLE_ENTERPRISE_ACCESS_KEY: ${{ secrets.GE_ACCESS_TOKEN }}
+  GRADLE_ENTERPRISE_CACHE_USERNAME: ${{ secrets.GE_CACHE_USERNAME }}
+  GRADLE_ENTERPRISE_CACHE_PASSWORD: ${{ secrets.GE_CACHE_PASSWORD }}
+
+jobs:
+  beam_PostCommit_TransformService_Direct:
+    if: |
+      github.event_name == 'workflow_dispatch' ||
+      github.event_name == 'schedule' ||
+      github.event.comment.body == 'Run TransformService_Direct PostCommit'
+    runs-on: [self-hosted, ubuntu-20.04, main]
+    timeout-minutes: 120
+    name: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+    strategy:
+      matrix:
+        job_name: ["beam_PostCommit_TransformService_Direct"]
+        job_phrase: ["Run TransformService_Direct PostCommit"]
+    steps:
+      - uses: actions/checkout@v3
+      - name: Setup repository
+        uses: ./.github/actions/setup-action
+        with:
+          comment_phrase: ${{ matrix.job_phrase }}
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          github_job: ${{ matrix.job_name }} (${{ matrix.job_phrase }})
+      - name: Set up Java 11
+        uses: actions/setup-java@v3.11.0
+        with:
+          distribution: 'temurin'
+          java-version: '11'
+      - name: Install Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: '3.8'
+      - name: run TransformService Direct script
+        uses: ./.github/actions/gradle-command-self-hosted-action
+        with:
+          gradle-command: :sdks:python:test-suites:direct:xlang:transformServicePythonUsingJava
+          arguments: |
+            -PcompileAndRunTestsWithJava11 \
+            -Pjava11Home=$JAVA_HOME_11_X64 \
+            -PuseWheelDistribution \
+            -PpythonVersion=3.8 \
```

Comment:
It seems like we need to run this Gradle task for two Python versions. You can check the log here: https://ci-beam.apache.org/view/PostCommit/job/beam_PostCommit_TransformService_Direct/lastSuccessfulBuild/consoleFull

Or you can look at the Groovy file. There is the code where the steps are executed for the list of CROSS_LANGUAGE_VALIDATES_RUNNER_PYTHON_VERSIONS. This list consists of two elements: LOWEST_SUPPORTED and HIGHEST_SUPPORTED. The code can be found in the PythonTestProperties class, specifically on lines 33 and 37.

Finally, we need to add the same step but for a different version of Python.

Reply:
Thanks, this is very helpful, will add python 3.11

Reply:
Fixed